CRM/Contribute/Form/UpdateBilling.php

```diff
@@ -202,28 +201,29 @@ public function postProcess() {
     $processorParams['country'] = CRM_Core_PseudoConstant::country($params["billing_country_id-{$this->_bltID}"], FALSE);
     $processorParams['month'] = CRM_Core_Payment_Form::getCreditCardExpirationMonth($processorParams);
     $processorParams['year'] = CRM_Core_Payment_Form::getCreditCardExpirationYear($processorParams);
-    $processorParams['subscriptionId'] = $this->getSubscriptionDetails()->processor_id;
-    $processorParams['amount'] = $this->_subscriptionDetails->amount;
+    $processorParams['recurProcessorID'] = $processorParams['subscriptionId'] = $this->getSubscriptionDetails()->processor_id;
+    $processorParams['amount'] = $this->getSubscriptionDetails()->amount;
+    $processorParams['contributionRecurID'] = $processorParams['id'] = $this->getContributionRecurID();
```

Comment:
Do we need both? If not future us might be trying to chase our tail making sense of whether `id` is required  so I'd only add `id` if there is a reason since it's a duplicate

Reply:
No. I just added it for consistency. I've written helpers to convert params to propertyBag in a version-safe/compatible way in the mjwshared library - passing `contributionRecurID` actually simplifies that helper because we just check if it has that property and if not fallback to retrieving from `subscriptionId`.  Updated